django_ratelimit/core.py

```diff
@@ -227,45 +248,99 @@ def get_usage(request, group=None, fn=None, key=None, rate=None, method=ALL,
     cache_key = _make_cache_key(group, window, rate, value, method)
 
     count = None
-    try:
-        added = cache.add(cache_key, initial_value, period + EXPIRATION_FUDGE)
-    except socket.gaierror:  # for redis
-        added = False
-    if added:
-        count = initial_value
+    if is_async:
+        async def inner():
+            try:
+                # Some caches don't have an async implementation
```

Comment:
It looks like the `aadd` methods are defined in `BaseCache` (see the next comment down for the link) so in theory any cache should have access to them

Reply:
I've tried it before, and if someone is using an old cache this can fail. This is just to support caches that may not define the cache interface with those methods. Test implementations can also be missing this.

Reply:
Can you clarify what you mean by "old cache"? Is it an implementation that doesn't inherit from `BaseCache`?